cmd/agent/common/jsonpatch/jsonpatch.go

```diff
@@ -0,0 +1,312 @@
+package jsonpatch
```

Comment:
Yep. What is this? Why do we need it? Can't we avoid it by just using `encoding/json`?

Reply:
Thank you for the feedback.
Unfortunately neither [evanphx/json-patch](https://github.com/evanphx/json-patch) nor [mattbaird/jsonpatch](https://github.com/mattbaird/jsonpatch) have an interface that allows for the creation of a JSONPatch incrementally.

I was not able to find a public implementation that would allow me to accumulate "add" operations and yield a JSONPatch list of operations at the end.

This code here was extracted from our [internal webhook (private link)](https://github.com/DataDog/k8s-admission-webhooks/blob/master/pkg/jsonpatch/jsonpatch.go).
We do not need all methods, but we do need a subset of it. I imagined that this file was extracted and modified from a public source, but I was not able to find any public libraries with the content of this file, so I'm assuming it was mostly created internally by us.

We could have as a goal to add the ability to incrementally build a JSONPatch to one of the exiting popular libraries, but that would be a stretch goal.

Reply:
Did we write it at Datadog? Honestly it doesn't look like it. I think it's copied from somewhere else. If no, then that's fine. We might need copyright headers in this case. If it's indeed copied from somewhere, please add a comment at the top followed by an empty line, something like:
```go
// Copied from github.com/......

package jsonpatch
```